src/util/helpers.ts

```diff
@@ -127,6 +128,19 @@ function fetchJSON(url, opts, cb) {
     .catch((err) => cb(err))
 }
 
+const resolveAfter = delay => new Promise(ok => setTimeout(ok, delay));
+
+function throttle(fn, delay) {
+    let wait: any = Promise.resolve();
+    return (...args) => {
+      const res = wait.then(() => fn(...args));
+      wait = wait.then(() => resolveAfter(delay));
+      return res;
+    };
+  }
```

Comment:
This seems like a lot of code for `setTimeout(() => { thing() }, X)`

Reply:
A simple `setTimeout` would just offset when each request is called by 5 seconds, instead of preventing the rate limit. If you called it 10x in a row, the system would wait 5 seconds, then fire them all at basically the same time.

This `throttle` function keeps the state across requests, so any time `throttledFetchJSON` is called it puts into the queue, so to speak. You can call it 10x in a row, and it will fire each time 5 seconds apart. 

This also doesn't slow down the first time it's requested, since we aren't at risk of being rate limited with our first request.

Reply:
Ohh interesting